### PR TITLE
🧹 providers coordinator v2

### PR DESCRIPTION
**Pull request by imilchev**
The goal is to allow creating local coordinators with a parent coordinator. The way the code is setup is:
- providers are always managed by the parent coordinator. This means eventually all providers should end up in the global coordinator
- the local coordinator only keeps a reference to the started providers but are not managing them
- from the perspective of the global coordinator any provider started by a local coordinator is ephemeral. This means the local coordinators have dedicated providers for them. This allows scans to run independently of each other
- there is no direct dependency from the global coordinator to its children

### CODE REVIEW

File: explorer/scan/discovery.go [hunk 1]
```diff
@@ -30,13 +36,26 @@ type AssetWithError struct {
 
 type DiscoveredAssets struct {
 	platformIds map[string]struct{}
-	Assets      []*AssetWithRuntime
-	Errors      []*AssetWithError
+	// Assets      []*AssetWithRuntime
```
Comment (jaym) [inline explorer/scan/discovery.go:39]: should this line be removed

File: explorer/scan/discovery.go [hunk 2]
```diff
@@ -30,13 +36,26 @@ type AssetWithError struct {
...
+	Errors []*AssetWithError
+
+	RootAssets map[*inventory.Asset]*RootAsset
+}
+
+func (d *DiscoveredAssets) AddRoot(root *inventory.Asset, coordinator providers.Coordinator, runtime *providers.Runtime) bool {
```
Comment (jaym) [inline explorer/scan/discovery.go:45]: runtime isn't used. is that expected?

File: explorer/scan/discovery.go [hunk 3]
```diff
@@ -146,15 +184,15 @@ func DiscoverAssets(ctx context.Context, inv *inventory.Inventory, upstream *ups
 	// if there is exactly one asset, assure that the --asset-name is used
 	// TODO: make it so that the --asset-name is set for the root asset only even if multiple assets are there
 	// This is a temporary fix that only works if there is only one asset
-	if len(discoveredAssets.Assets) == 1 && invAssets[0].Name != "" && invAssets[0].Name != discoveredAssets.Assets[0].Asset.Name {
-		log.Debug().Str("asset", discoveredAssets.Assets[0].Asset.Name).Msg("Overriding asset name with --asset-name flag")
-		discoveredAssets.Assets[0].Asset.Name = invAssets[0].Name
-	}
+	// if len(discoveredAssets.Assets) == 1 && invAssets[0].Name != "" && invAssets[0].Name != discoveredAssets.Assets[0].Asset.Name {
```
Comment (jaym) [inline explorer/scan/discovery.go:187]: commented out code

**Comment by arlimus** [comment]
Thank you for taking a stab at this! We are motivated by random provider shutdowns across the system and by supporting multiple scans at the same time. We'll have a separate call to go over a discussion of this, so we will keep this PR as draft until then.

**Comment by imilchev** [comment]
we are fixing this by getting rid of ephemeral providers and making sure we have a more robust provider management system. Closing this one

**Pull request closed without merging**